go/vt/vttablet/tabletserver/throttle/throttler.go

```diff
@@ -518,6 +520,43 @@ func (throttler *Throttler) Operate(ctx context.Context) {
 	}
 }
 
+func (throttler *Throttler) generateTabletHTTPProbeFunction(ctx context.Context, clusterName string, probe *mysql.Probe) (probeFunc func() *mysql.MySQLThrottleMetric) {
+	if probe.TabletHost == "" {
+		// nil function means no override; throttler will use default probe behavior, which is to open a direct
+		// connection to mysql and run a query
+		return nil
+	}
+	return func() *mysql.MySQLThrottleMetric {
+		// Hit a tablet's `check-self` via HTTP, and convert its CheckResult JSON output into a MySQLThrottleMetric
+		mySQLThrottleMetric := mysql.NewMySQLThrottleMetric()
+		mySQLThrottleMetric.ClusterName = clusterName
+		mySQLThrottleMetric.Key = probe.Key
+
+		tabletCheckSelfURL := fmt.Sprintf("http://%s:%d/throttler/check-self?app=vitess", probe.TabletHost, probe.TabletPort)
+		resp, err := throttler.httpClient.Get(tabletCheckSelfURL)
+		if err != nil {
+			mySQLThrottleMetric.Err = err
+			return mySQLThrottleMetric
+		}
+		b, err := io.ReadAll(resp.Body)
+		if err != nil {
+			mySQLThrottleMetric.Err = err
+			return mySQLThrottleMetric
+		}
+		checkResult := &CheckResult{}
+		if err := json.Unmarshal(b, checkResult); err != nil {
+			mySQLThrottleMetric.Err = err
+			return mySQLThrottleMetric
+		}
+		mySQLThrottleMetric.Value = checkResult.Value
+
+		if checkResult.StatusCode == http.StatusInternalServerError {
```

Comment:
Would we ever get here? I would think we would have gone into one of the earlier err blocks. Doesn't hurt, but if we do handle http codes we could handle any of the err codes, 4XX and 5XX or even anything not 2XX.

Reply:
Probably 5XX would be handled earlier and still I'd want to surface that. As for the other error codes, these are app-level codes that are generated by the underlying check. There is no need to propagate the error to the upper level, it's enough that we propagate the `Value`. In fact, it hurts if we report back an error, where there is no error per-se, just an app-level decision that "this check is a no-go". It's the upper layer, the collector's responsibility to collect all values and decide if the total state is "go" or "no-go".
The introduction of this mid HTTP layer complicates things because we now only have implicit/secondary understanding of the nature of errors on the MySQL side.
As it stands, the current logic passes all the tests correctly, which means it complies with existing behavior.